enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/storecopy/edge/StoreCopyClient.java

```diff
@@ -58,4 +60,19 @@ public long copyStoreFiles( CoreMember from, StoreFileStreams storeFileStreams )
             coreClient.removeStoreFileStreamingCompleteListener( fileStreamingCompleteListener );
         }
     }
+
+    public StoreId fetchStoreId( CoreMember from ) throws StoreIdDownloadFailedException
+    {
+        CompletableFuture<StoreId> storeIdCompletableFuture = new CompletableFuture<>();
+        coreClient.setStoreIdConsumer( storeIdCompletableFuture::complete );
+        coreClient.requestStoreId( from );
+        try
+        {
+            return storeIdCompletableFuture.get();
+        }
+        catch ( InterruptedException | ExecutionException e )
```

Comment:
It is probably a good idea to restore interruption status for `InterruptedException`